staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -248,17 +267,27 @@ func (t *envelopeTransformer) doEncode(request *kmstypes.EncryptedObject) ([]byt
 	if err := validateEncryptedObject(request); err != nil {
 		return nil, err
 	}
-	return proto.Marshal(request)
+
+	buffer := t.encryptedObjectDecoder.bufferPool.Get().(*proto.Buffer)
+	defer t.encryptedObjectDecoder.bufferPool.Put(buffer)
+
+	if err := buffer.Marshal(request); err != nil {
```

Comment:
Add a unit test to catch this bug.

```suggestion

	buffer.Reset()
	if err := buffer.Marshal(request); err != nil {
```